src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -357,11 +363,41 @@ public Stream Open()
                     return OpenInWriteMode();
                 case ZipArchiveMode.Update:
                 default:
-                    Debug.Assert(_archive.Mode == ZipArchiveMode.Update);
                     return OpenInUpdateMode();
             }
         }
 
+
+        /// <summary>
+        /// Opens the entry. If the archive that the entry belongs to was opened in Read mode, the returned stream will be readable, and it may or may not be seekable. If Create mode, the returned stream will be writable and not seekable. If Update mode, the returned stream will be readable, writable, seekable, and support SetLength.
+        /// </summary>
+        /// <returns>A Stream that represents the contents of the entry.</returns>
+        /// <exception cref="IOException">The entry is already currently open for writing. -or- The entry has been deleted from the archive. -or- The archive that this entry belongs to was opened in ZipArchiveMode.Create, and this entry has already been written to once.</exception>
+        /// <exception cref="InvalidDataException">The entry is missing from the archive or is corrupt and cannot be read. -or- The entry has been compressed using a compression method that is not supported.</exception>
+        /// <exception cref="ObjectDisposedException">The ZipArchive that this entry belongs to has been disposed.</exception>
+        public Stream Open(string? password = null, EncryptionMethod encryptionMethod = EncryptionMethod.ZipCrypto)
```

Comment:
Consider also how adding new Open overload interacts with https://github.com/dotnet/runtime/issues/101243,

Also, I believe Jeremy expressed strong opinions that we should not select default encryption method and require it be explicitly specified by the user.